### ISSUE TITLE
Adguard Extension randomly crashes in Chrome on launch

### ISSUE THREAD

**Issue by scrutinizer11** [AdguardTeam/AdguardBrowserExtension#2165]
### Issue Details

- **AdGuard version:**
  - AdGuard Extension v4.1.1 
- **Browser and version:**
  - Chromium 104.0.5112.79 (based on Chrome 104) 
- **Operating system and version:**
   macOS 10.7, 10.9
- **Filters you use in AdGuard:**
   - AdGuard Annoyances filter
   - Adblock Warning Removal List
   - Web Annoyances Ultralist
   - I don't care about cookies
   - EasyList Cookie List
   - AdGuard URL Tracking filter
   - AdGuard Tracking Protection filter
   - AdGuard Base filter
   - EasyList
   - NoCoin Filter List
   - AdGuard Russian filter
   - EasyList Italy
   - EasyList Lithuania
   - RU AdList: Counters
   
- **Stealth Mode settings:**
  **_Off_**
- **Other browser extensions:**
  - Stylus
  - DNS Checker
  - File Converter - By Online-Convert.com
  - iCloud Dashboard
  - Grammarly
  - Mate Translate
  - Wayback Machine
  - Dark Reader
  - Youtube Enhancer

### Additional Information

Sometimes, but often enough to get a noticeable effect, Adguard Extension crashes when I launch Chromium. Chromium sends a notification about that telling me to reload it.

**Comment by Birbber** [comment]
@scrutinizer11 Hi, I wasn't able to reproduce the issue with the same extensions in chrome 104. 
I suspect some of these extensions might affect the work of AdGuard. Could you try disabling some of them and monitor the issue?
Also a screenshot of the notification would be helpful.

**Comment by zzebrum** [comment]
@scrutinizer11 any news?

**Comment by scrutinizer11** [comment]
No news. Still crashes. It's nothing special to show other than the notification _Adguard has crashed. Click on this bubble to reload it. That's it._ It can happen several times in a row during a short interval.

**Comment by scrutinizer11** [comment]
One tab was playing YouTube content, another one was an online forum. The extension crashed more than once and the active tab (the forum) get rid of its contents, it became blank. 
The extension's management pane displayed the error
> Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.

The error pointed to this code:

[Adguard-error.log](https://github.com/AdguardTeam/AdguardBrowserExtension/files/9437315/Adguard-error.log)

**Comment by maximtop** [comment]
@scrutinizer11 I didn't find any errors in the code you sent. The error you are referring can't be the cause of crashing. Can you share your extension settings and the websites open in the tabs?

**Comment by scrutinizer11** [comment]
I didn't say the code contains the error. I said that the quoted error **pointed** to the code snippet that showed up on the extension's management pane.

I enclose a JSON file with the settings, as you requested.  As to the tab configuration, I have it varying frequently so it's difficult to recall the exact pattern at which the tabs turn up at the time of the bug cropping up because the crashes occur out of the blue. I did notice though, that for the most part, they happen when one or several tabs with resource-hogging content are opened. Usually, it's Facebook and Youtube among others. Open any number of tabs above 2 including any demanding site.
[20220829_080824_adg_ext_settings_4.1.1.txt](https://github.com/AdguardTeam/AdguardBrowserExtension/files/9441740/20220829_080824_adg_ext_settings_4.1.1.txt)

**Comment by scrutinizer11** [comment]
> Cannot load extension with file or directory name _metadata. Filenames starting with "_" are reserved for use by the system.

[ag-warning-30·08·2022 17·05.txt](https://github.com/AdguardTeam/AdguardBrowserExtension/files/9453518/ag-warning-30.08.2022.17.05.txt)

**Comment by scrutinizer11** [comment]
Now it crashes like crazy on every Chromiums launch, every tab opening (even this page).

**Comment by maximtop** [comment]
Can you describe the steps to reproduce the error in the last comment?

**Comment by scrutinizer11** [comment]
Just launch the browser with several tabs or open them after the launch. Some of those have to be opened on YouTube, Facebook or a similar site. It's a good chance it may then kick off the trigger setting a crash cycle in motion. In some cases, even opening sites such as GitHub leads to the crash as well.
Better yet, use a machine from 2012 with any Mac operating system.

**Comment by maximtop** [comment]
Are the settings the same as you've sent before?

**Comment by maximtop** [comment]
@scrutinizer11 I've asked QA to test on Mohave 10.14, but they could not reproduce crashes.

**Comment by scrutinizer11** [comment]
I have discovered something accidentally but am not sure if this is related and what it means. I clicked on _Inspect Views background.js:1_ just upon reloading the extension after it crashed: it opened the referred "view" and under _Issues_ I saw this:

![Screen Shot 2022-09-14 at 02 41 00](https://user-images.githubusercontent.com/45983734/190028585-bb8c454c-54fa-4dbd-91cf-138dd95322fa.png)

Is it in any way helpful?

**Comment by maximtop** [comment]
@scrutinizer11 we will remove usage of described api, but it is unlikely to be the cause of the crashes

**Comment by scrutinizer11** [comment]
![ADG-error_2022-09-25_02-41-46-660](https://user-images.githubusercontent.com/45983734/192122449-02c57a51-f2c7-40aa-92bf-219103c32cab.png)

**Comment by maximtop** [comment]
> ![ADG-error_2022-09-25_02-41-46-660](https://user-images.githubusercontent.com/45983734/192122449-02c57a51-f2c7-40aa-92bf-219103c32cab.png)

Have you reloaded the extension? This error may occur if the extension background page was reloaded while the content script was sending messages.

**Comment by scrutinizer11** [comment]
Yes, I reloaded it after it crashed. The thing crashes a hell of a lot, at the beginning and the middle of the browsing session.

**Comment by krystian3w** [comment]
more time gh actions

**Comment by krystian3w** [comment]
wytyfy?

**Comment by mityaua** [comment]
Still actual - AdGuard AdBlocker (Beta) 5.2.500.0

<img width="1357" height="57" alt="Image" src="https://github.com/user-attachments/assets/5d6b5dca-6212-4198-aa6d-c855758dfc7b" />
